modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/ServiceConfigurationUpdateHandler.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileCopyrightText: Huawei Inc.
+ *
+ */
+
+package org.eclipse.xpanse.modules.deployment;
+
+import jakarta.annotation.Resource;
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.LockModeType;
+import jakarta.transaction.Transactional;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.UUID;
+import lombok.extern.slf4j.Slf4j;
+import org.eclipse.xpanse.modules.database.serviceconfiguration.update.ServiceConfigurationUpdateRequest;
+import org.eclipse.xpanse.modules.database.serviceconfiguration.update.ServiceConfigurationUpdateStorage;
+import org.eclipse.xpanse.modules.models.serviceconfiguration.enums.ServiceConfigurationStatus;
+import org.hibernate.exception.LockTimeoutException;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+/**
+ * Service configuration more detailed request handler.
+ */
+@Slf4j
+@Component
+public class ServiceConfigurationUpdateHandler {
+
+    public static final String JAVAX_PERSISTENCE_LOCK_TIMEOUT = "javax.persistence.lock.timeout";
+
+    @Value("${javax.persistence.lock.timeout:2000}")
+    private int lockTimeout;
+
+    @Resource
+    private EntityManager entityManager;
+
+    @Resource
+    private ServiceConfigurationUpdateStorage serviceConfigurationUpdateStorage;
+
+    /**
+     * Update service configuration update request.
+     */
+    @Transactional
+    public ServiceConfigurationUpdateRequest
+            updateServiceConfigurationUpdateRequest(UUID id, String resourceName) {
+        try {
+            Map<String, Object> properties = new HashMap<>();
+            properties.put(JAVAX_PERSISTENCE_LOCK_TIMEOUT, lockTimeout);
+            ServiceConfigurationUpdateRequest request = entityManager
+                    .find(ServiceConfigurationUpdateRequest.class, id,
+                            LockModeType.PESSIMISTIC_WRITE, properties);
+            if (Objects.isNull(request.getResourceName())) {
+                request.setResourceName(resourceName);
+            }
+            request.setStatus(ServiceConfigurationStatus.PROCESSING);
+            return serviceConfigurationUpdateStorage.storeAndFlush(request);
+        } catch (LockTimeoutException e) {
+            log.error("Update service configuration update request object timed out", e);
+            return null;
```

Comment:
If the method is important to the service, please throw the `LockTimeoutException` and use  `@Retryable(value = LockTimeoutException.class, maxAttempts = 3)` on this method.

Reply:
@baixinsui 
![image](https://github.com/user-attachments/assets/f0a4d545-5fef-42d6-90dc-83de1cf48650)
There is no need for a retry mechanism here. For parallel requests, if the lock times out when obtaining data, 204 needs to be returned.